L1Trigger/TrackFindingTracklet/src/PurgeDuplicate.cc

```diff
@@ -86,6 +87,11 @@ void PurgeDuplicate::execute(std::vector<Track>& outputtracks, unsigned int iSec
   inputstublists_.clear();
   mergedstubidslists_.clear();
 
+  sortedinputtracklets_.clear();
```

Comment:
I've never understood why all these vectors are declared as class data members. If they were instead declared inside the loop over bins, e.g. near here https://github.com/cms-L1TK/cmssw/blob/dally_CMFix/L1Trigger/TrackFindingTracklet/src/PurgeDuplicate.cc#L140 , they would automatically be deleted and recreated between loops, so no clear would be needed. Am I missing something?

Reply:
No, I just wanted to mirror the actual class data members, but I can change them to only be declared in the loop over bins